sup3r/models/abstract.py

```diff
@@ -1309,6 +1260,10 @@ def generate(self,
             (n_obs, spatial_1, spatial_2, n_features)
             (n_obs, spatial_1, spatial_2, n_temporal, n_features)
         """
+        if (isinstance(exogenous_data, dict)
+                and not isinstance(exogenous_data, ExoData)):
+            exogenous_data = ExoData(exogenous_data)
```

Comment:
I don't think you need the double `isinstance()` call do you? Or is this just because ExoData inherits dict? Either way, just use the more restrictive of the two conditions. 

Also, since this is just two lines, why not just put this everywhere in this file that accepts exogenous_data input so it can always be dict/exodata/none